be/src/types/checker/type_checker.cpp

```diff
@@ -0,0 +1,319 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "types/checker/type_checker.h"
+
+#include <fmt/format.h>
+
+namespace starrocks {
+// java.lang.Byte
+
+StatusOr<LogicalType> ByteTypeChecker::check(const std::string& java_class,
+                                             const SlotDescriptor* slot_desc) const {
+    auto type = slot_desc->type().type;
+    if (type != TYPE_BOOLEAN && type != TYPE_TINYINT && type != TYPE_SMALLINT && type != TYPE_INT &&
+        type != TYPE_BIGINT) {
+        return Status::NotSupported(
+                fmt::format("Type mismatches on column[{}], JDBC result type is Byte, please set the type to "
+                            "one of boolean,tinyint,smallint,int,bigint",
+                            slot_desc->col_name()));
+    }
+    if (type == TYPE_BOOLEAN) {
+        return TYPE_BOOLEAN;
+    }
+    return TYPE_TINYINT;
+}
+
+// com.clickhouse.data.value.UnsignedByte
+
```

Comment:
useless empty line.

Reply:
I will clean up the code and comments later